video/filter/vf_vapoursynth.c

```diff
@@ -210,13 +211,16 @@ static void copy_mp_to_vs_frame_props_map(struct priv *p, VSMap *map,
 }
 
 static int set_vs_frame_props(struct priv *p, VSFrameRef *frame,
-                              struct mp_image *img, int dur_num, int dur_den)
+                              struct mp_image *img, int dur_num, int dur_den, int frameno)
 {
     VSMap *map = p->vsapi->getFramePropsRW(frame);
     if (!map)
         return -1;
     p->vsapi->propSetInt(map, "_DurationNum", dur_num, 0);
     p->vsapi->propSetInt(map, "_DurationDen", dur_den, 0);
+    p->vsapi->propSetInt(map, "__frameno", frameno, 0);
+    if(img->dovi && img->dovi->size)
+        p->vsapi->propSetData(map,"_DoVi", img->dovi->data, img->dovi->size, 0);
```

Comment:
also why set this as a prop? can it not be copied from `p->buffered[idx]` below?

Reply:
so the Vapoursynth script could use these data

Reply:
Is `_DoVi` name commonly used by other VS hosts or it is our invention? Probably would be good to use same name as it may be used already.

Reply:
I believe there're no other VS hosts dealing with DoVi data.

Reply:
Only the RPU binary already uses the `DolbyVisionRpu` prop.

Reply:
Is the any reasonable way that a VS script could actually use this data? like a python parser for dovi that exists?

Reply:
frame interpolation script wants to know if DoVi data in the adjacent frames are the same or different

Reply:
If `img->dovi` actually holds `pl_dovi_metadata`, then there's probably nothing that would use it.
It doesn't need parsing since it's a C struct.

Reply:
> frame interpolation script wants to know if DoVi data in the adjacent frames are the same or different

Then it should compare the `DolbyVisionRPU` prop.

Reply:
FWIW the point of my question was the following: if there is no reasonable way a VS script can make use of the DoVI data then there is no need to set it as a property at all.

Reply:
> if there is no reasonable way a VS script can make use of the DoVI data

My use case is this: the script can _compare_ raw DoVi bytes between two frames w/o parsing them. If the data are different then merging those frames (in any way) is a bad idea.

Reply:
Okay then, good enough.
This PR still needs a rebase however.